test/integration/controllermanager/quota/quota_test.go

```diff
@@ -0,0 +1,162 @@
+			Provider: &gardencorev1beta1.SecretBindingProvider{
+				Type: providerType,
+			},
+			SecretRef: corev1.SecretReference{
+				Name:      resourceName,
+				Namespace: testNamespace.Name,
+			},
+			Quotas: []corev1.ObjectReference{
+				{
+					Name:      resourceName,
+					Namespace: testNamespace.Name,
+				},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		By("Create Secret")
+		Expect(testClient.Create(ctx, secret)).To(Succeed())
+		log.Info("Created Secret for test", "secret", client.ObjectKeyFromObject(secret))
+
+		By("Create Quota")
+		Expect(testClient.Create(ctx, quota)).To(Succeed())
+		log.Info("Created Quota for test", "quota", client.ObjectKeyFromObject(quota))
+
+		DeferCleanup(func() {
+			By("Delete Quota")
+			Expect(testClient.Delete(ctx, quota)).To(Or(Succeed(), BeNotFoundError()))
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(quota), quota)
+			}).Should(BeNotFoundError())
+		})
+
+		DeferCleanup(func() {
+			By("Delete Secret")
+			Expect(testClient.Delete(ctx, secret)).To(Or(Succeed(), BeNotFoundError()))
+		})
+
+		if secretBinding != nil {
+			By("Create SecretBinding")
+			Expect(testClient.Create(ctx, secretBinding)).To(Succeed())
+			log.Info("Created SecretBinding for test", "secretBinding", client.ObjectKeyFromObject(secretBinding))
+
+			DeferCleanup(func() {
+				By("Delete SecretBinding")
+				Expect(testClient.Delete(ctx, secretBinding)).To(Or(Succeed(), BeNotFoundError()))
+			})
+		}
+	})
+
+	Context("no SecretBinding referencing Quota", func() {
+		BeforeEach(func() {
+			secretBinding = nil
+		})
+
+		It("should add the finalizer and release it on deletion", func() {
+			By("Ensure finalizer got added")
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, objectKey, quota)).To(Succeed())
+				g.Expect(quota.Finalizers).To(ConsistOf("gardener"))
+			}).Should(Succeed())
+
+			By("Delete Quota")
+			Expect(testClient.Delete(ctx, quota)).To(Succeed())
+
+			By("Ensure Quota is released")
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(quota), quota)
+			}).Should(BeNotFoundError())
+		})
+	})
+
+	Context("SecretBinding referencing Quota", func() {
+		JustBeforeEach(func() {
+			By("Ensure finalizer got added")
+			Eventually(func(g Gomega) {
+				g.Expect(testClient.Get(ctx, objectKey, quota)).To(Succeed())
+				g.Expect(quota.Finalizers).To(ConsistOf("gardener"))
+			}).Should(Succeed())
+
+			By("Delete Quota")
+			Expect(testClient.Delete(ctx, quota)).To(Succeed())
+		})
+
+		It("should add finalizer and not release it on deletion since there is still referencing SecretBinding", func() {
+			By("Ensure Quota is not released")
+			Consistently(func() error {
+				return testClient.Get(ctx, objectKey, quota)
+			}).Should(Succeed())
+		})
+
+		It("should add the finalizer and release it on deletion after SecretBinding got deleted", func() {
+			By("Delete Secretbinding")
```

Comment:
This is still not resolved
```suggestion
			By("Delete SecretBinding")
```